commit 028b4b21df26fee67b3ce75c6f14fcfd3c7cf2ee
Author: Michael Paquier <michael@paquier.xyz>
Date:   Sun Aug 4 03:07:20 2024 +0900

    Fix incorrect format placeholders in pgstat.c
    
    These should have been switched from %d to %u in 3188a4582a8c in the
    debugging elogs added in ca1ba50fcb6f.  PgStat_Kind should never be
    higher than INT32_MAX, but let's be clean.
    
    Issue noticed while hacking more on this area.

diff --git a/src/backend/utils/activity/pgstat.c b/src/backend/utils/activity/pgstat.c
index 143478aca0f..2e13670fc1f 100644
--- a/src/backend/utils/activity/pgstat.c
+++ b/src/backend/utils/activity/pgstat.c
@@ -1596,31 +1596,31 @@ pgstat_read_statsfile(XLogRecPtr redo)
 						goto error;
 					}
 
 					if (!pgstat_is_kind_valid(kind))
 					{
-						elog(WARNING, "invalid stats kind %d for entry of type %c",
+						elog(WARNING, "invalid stats kind %u for entry of type %c",
 							 kind, t);
 						goto error;
 					}
 
 					info = pgstat_get_kind_info(kind);
 
 					if (!info->fixed_amount)
 					{
-						elog(WARNING, "invalid fixed_amount in stats kind %d for entry of type %c",
+						elog(WARNING, "invalid fixed_amount in stats kind %u for entry of type %c",
 							 kind, t);
 						goto error;
 					}
 
 					/* Load back stats into shared memory */
 					ptr = ((char *) shmem) + info->shared_ctl_off +
 						info->shared_data_off;
 
 					if (!read_chunk(fpin, ptr, info->shared_data_len))
 					{
-						elog(WARNING, "could not read data of stats kind %d for entry of type %c with size %u",
+						elog(WARNING, "could not read data of stats kind %u for entry of type %c with size %u",
 							 kind, t, info->shared_data_len);
 						goto error;
 					}
 
 					break;
@@ -1643,11 +1643,11 @@ pgstat_read_statsfile(XLogRecPtr redo)
 							goto error;
 						}
 
 						if (!pgstat_is_kind_valid(key.kind))
 						{
-							elog(WARNING, "invalid stats kind for entry %d/%u/%u of type %c",
+							elog(WARNING, "invalid stats kind for entry %u/%u/%u of type %c",
 								 key.kind, key.dboid, key.objoid, t);
 							goto error;
 						}
 					}
 					else
@@ -1662,36 +1662,36 @@ pgstat_read_statsfile(XLogRecPtr redo)
 							elog(WARNING, "could not read stats kind for entry of type %c", t);
 							goto error;
 						}
 						if (!read_chunk_s(fpin, &name))
 						{
-							elog(WARNING, "could not read name of stats kind %d for entry of type %c",
+							elog(WARNING, "could not read name of stats kind %u for entry of type %c",
 								 kind, t);
 							goto error;
 						}
 						if (!pgstat_is_kind_valid(kind))
 						{
-							elog(WARNING, "invalid stats kind %d for entry of type %c",
+							elog(WARNING, "invalid stats kind %u for entry of type %c",
 								 kind, t);
 							goto error;
 						}
 
 						kind_info = pgstat_get_kind_info(kind);
 
 						if (!kind_info->from_serialized_name)
 						{
-							elog(WARNING, "invalid from_serialized_name in stats kind %d for entry of type %c",
+							elog(WARNING, "invalid from_serialized_name in stats kind %u for entry of type %c",
 								 kind, t);
 							goto error;
 						}
 
 						if (!kind_info->from_serialized_name(&name, &key))
 						{
 							/* skip over data for entry we don't care about */
 							if (fseek(fpin, pgstat_get_entry_len(kind), SEEK_CUR) != 0)
 							{
-								elog(WARNING, "could not seek \"%s\" of stats kind %d for entry of type %c",
+								elog(WARNING, "could not seek \"%s\" of stats kind %u for entry of type %c",
 									 NameStr(name), kind, t);
 								goto error;
 							}
 
 							continue;
@@ -1709,11 +1709,11 @@ pgstat_read_statsfile(XLogRecPtr redo)
 
 					/* don't allow duplicate entries */
 					if (found)
 					{
 						dshash_release_lock(pgStatLocal.shared_hash, p);
-						elog(WARNING, "found duplicate stats entry %d/%u/%u of type %c",
+						elog(WARNING, "found duplicate stats entry %u/%u/%u of type %c",
 							 key.kind, key.dboid, key.objoid, t);
 						goto error;
 					}
 
 					header = pgstat_init_entry(key.kind, p);
@@ -1721,11 +1721,11 @@ pgstat_read_statsfile(XLogRecPtr redo)
 
 					if (!read_chunk(fpin,
 									pgstat_get_entry_data(key.kind, header),
 									pgstat_get_entry_len(key.kind)))
 					{
-						elog(WARNING, "could not read data for entry %d/%u/%u of type %c",
+						elog(WARNING, "could not read data for entry %u/%u/%u of type %c",
 							 key.kind, key.dboid, key.objoid, t);
 						goto error;
 					}
 
 					break;
